app/assets/javascripts/components/Nav.js

```diff
@@ -0,0 +1,283 @@
+
+        self.timeout = window.setTimeout(function() {
+          self._closeDesktopLevel2();
+        }, self.delay);
+      }
+    });
+
+  this.$navLevel_1_Heading
+    .mouseenter(function(e) {
+      if (self.viewportWidth >= self.smallViewport) {
+        window.clearTimeout(self.timeout);
+
+        self.timeout = window.setTimeout(function() {
+          self._openDesktopLevel2($(e.target));
+        }, self.delay);
+      }
+    })
+    .mousedown(function(e) {
+      if (self.viewportWidth >= self.smallViewport) {
+        e.preventDefault();
+      }
+    }).on('touchstart', function(e) {
+      if (self.viewportWidth >= self.smallViewport) {
+        // remove hover event handler if touch event detected
+        $(e.target).parents('[data-nav-level-1-heading]').off('mouseenter');
+      }
+    }).on('touchend', function(e) {
+      if (self.viewportWidth >= self.smallViewport) {
+        var index = $(e.target).parents('[data-nav-level-1-heading]');
+        // touch event on top level heading triggers open/close subnav
+        if ($(e.target).parents('[data-nav-level-1]').hasClass('is-active')) {
+          self._closeDesktopLevel2(index);
+        } else {
+          self._openDesktopLevel2(index);
+        }
+      }
+    });
+
+  this.$nav
+    .mouseleave(function() {
+      if (self.viewportWidth >= self.smallViewport) {
+        window.clearTimeout(self.timeout);
+
+        self.closeNavTimeout = window.setTimeout(function() {
+          self._closeDesktopLevel2();
+        }, self.delay);
+      }
+    })
+    .mouseenter(function() {
+      if (self.viewportWidth >= self.smallViewport) {
+        window.clearTimeout(self.closeNavTimeout);
+      }
+    });
+
+  this.$navLevel_2_Extended_Heading
+    .mouseenter(function() {
+      if (self.viewportWidth >= self.smallViewport) {
+        self._openDesktopLevel3(this);
+      }
+    });
+};
+
+/**
+ * Opens subnav on desktop
+ */
+Nav.prototype._openDesktopLevel2 = function(index) {
+  if (!this.$nav.hasClass('is-active')) {
+    this.$navLevel_3.removeClass('is-active');
+    this.$navLevel_2.removeClass('is-active');
+    this.$navLevel_1_item.removeClass('is-active');
+
+    $(index)
+      .parent('[data-nav-level-1-item]').addClass('is-active')
+      .siblings('[data-nav-level-2]').addClass('is-active');
+  }
+};
+
+/**
+ * Closes subnav on desktop
+ */
+Nav.prototype._closeDesktopLevel2 = function(index) {
+  if (!this.$nav.hasClass('is-active')) {
+    this.$navLevel_3.removeClass('is-active');
+    this.$navLevel_2.removeClass('is-active');
+    this.$navLevel_1_item.removeClass('is-active');
+
+    $(index)
+      .siblings('[data-nav-level-1-heading]')
+      .parents('[data-nav-level-1-item]')
+        .removeClass('is-active');
+  }
+};
+
+/**
+ * Opens level 3 on desktop
+ */
+Nav.prototype._openDesktopLevel3 = function(index) {
+  this.$navLevel_3.removeClass('is-active');
+  $(index).siblings('[data-nav-level-3').addClass('is-active');
```

Comment:
Typo here in `[data-nav-level-3` -> `[data-nav-level-3]`?

Reply:
Corrected (oops!)